xcube_stac/store.py

```diff
@@ -0,0 +1,410 @@
+        """
+        self.stac = stac
+
+    def get_open_data_params_schema(self, data_id: str = None) -> JsonObjectSchema:
+        """ Get the schema for the parameters passed as *open_params* to
+        :meth:`open_data`.
+
+        Args:
+            data_id (str, optional): An identifier of data that is provided by this
+                store. Defaults to None.
+
+        Returns:
+            JsonObjectSchema: containing the parameters used by the data opener
+                to open data.
+        """
+        # ToDo: to be adjusted
+        open_parms = {}
+        stac_schema = JsonObjectSchema(
+            properties=dict(**open_parms),
+            required=[],
+            additional_properties=False
+        )
+        return stac_schema
+
+    def open_data(self, data_id: str, **open_params) -> xr.Dataset:
+        """ Open the data given by the data resource identifier *data_id*
+        using the supplied *open_params*.
+
+        Args:
+            data_id (str): An identifier of data that is provided by this
+                store.
+
+        Returns:
+            xr.Dataset: An in-memory representation of the data resources
+                identified by *data_id* and *open_params*.
+        """
+        stac_schema = self.get_open_data_params_schema()
+        stac_schema.validate_instance(open_params)
+        return self.stac.open_data(data_id, **open_params)
+
+    def describe_data(
+        self, data_id: str, **open_params
+    ) -> DatasetDescriptor:
+        """ Get the descriptor for the data resource given by *data_id*.
+
+        Args:
+            data_id (str): An identifier of data that is provided by this
+                store.
+
+        Raises:
+            NotImplementedError: Not implemented yet.
+
+        Returns:
+            DatasetDescriptor: data descriptor containing meta data of
+                the data resources identified by *data_id*
+        """
+        # ToDo: implement describe_data method.
+        raise NotImplementedError("describe_data() operation is not supported yet")
+
+
+class StacDataStore(StacDataOpener, DataStore):
+    """ STAC implementation of the data store.
+    """
+
+    def __init__(self, **stac_kwargs):
+        """
+        Args:
+            **stac_kwargs: Parameters required by the STAC data store.
+                * url (str): URL to STAC catalog (required)
+                * collection_prefix (str): Path of collection used as
+                    entry point (optional)
+                * data_id_delimiter (str): Delimiter used to separate
+                    collections, items and assets from each other (optional)
+        """
+        super().__init__(stac=Stac(**stac_kwargs))
+
+    @classmethod
+    def get_data_store_params_schema(cls) -> JsonObjectSchema:
+        """ Get the JSON schema for instantiating a new data store.
+
+        Returns:
+            JsonObjectSchema: The JSON schema for the data store's parameters.
```

Comment:
```suggestion
        Returns:
            The JSON schema for the data store's parameters.
```
No need to repeat the type here.